source/portable/NetworkInterface/libslirp/MBuffNetifBackendLibslirp.c

```diff
@@ -472,6 +478,12 @@ static void vEnsurePollfdSize( SlirpBackendContext_t * pxCtx,
         if( pxCtx->pxPollFdArray == NULL )
         {
             pxCtx->pxPollFdArray = ( struct pollfd * ) malloc( xNewSize * sizeof( struct pollfd ) );
+
+            if( pxCtx->pxPollFdArray == NULL )
+            {
+                FreeRTOS_printf( ( "Failed to allocate memory for pxCtx->pxPollFdArray" ) );
```

Comment:
Note: Depending on the implementation, FreeRTOS_printf may not be usable from native threads.

Reply:
@tony-josi-aws  pls take note.

Reply:
Thanks @amazonKamath, in almost all of our +TCP demos, `FreeRTOS_printf` or `FreeRTOS_debug_printf` is mapped to put the print messages into a FreeRTOS queue thats monitored by a separate logging task to make it thread safe.